vllm/entrypoints/openai/api_server.py

```diff
@@ -629,6 +652,16 @@ async def fake_stream_generator() -> AsyncGenerator[str, None]:
                               tokenizer_mode=engine_args.tokenizer_mode,
                               trust_remote_code=engine_args.trust_remote_code)
 
+    if chat_template is not None:
+        tokenizer.chat_template = chat_template
+
+    tmp_template = tokenizer.chat_template or tokenizer.default_chat_template
+    if tmp_template:
+        logger.info(f"Chat template:\n{tmp_template}")
+    else:
+        logger.warning(
+            "No chat template loaded, the chat endpoint will be not work.")
+
```

Comment:
```suggestion
    if chat_template is not None:
        tokenizer.chat_template = chat_template

    final_template = tokenizer.chat_template or tokenizer.default_chat_template
    if final_template:
        logger.info(f"Using chat template:\n{tmp_template}")
    else:
        raise ValueError("No chat template loaded, the chat endpoint will be not work. There is no default template from the tokenizer. Please explicitly provide one through `--chat-template`."

```

I would prefer hard failing here to make sure there's a consistent experience as before when fastchat is not found. However this does mean if users want to use completion API only they won't be proceed forward. Perhaps we can add another flag to enable completion API with chat template?

Reply:
If you prefer a hard-fail, that's fine with me. I was just going with the approach of returning an error to the user, logging the error, and allowing the other endpoints to work because it seemed like a better user experience. Will go with a hard-fail instead.

Reply:
Oh maybe hard fail and `--completion-api-only` flag. What do you think? My thinking is that fast failing on the server side is better than users getting errors at runtime for Chat endpoint.

Reply:
I'll do an "endpoints" list as input, with an empty list meaning "all". I figure more endpoints will be added eventually, considering the OpenAI API expansion with agents and all that.

Any opposition?

Reply:
Sounds good!

Reply:
btw https://github.com/huggingface/transformers/blob/38e2633f80a4924bf613b0240622492beee4cfcc/src/transformers/tokenization_utils_base.py#L1731

Reply:
@simon-mo So hard-fail here is actually harder than I was hoping. We don't know that the template is broken until we go to apply it at runtime due to a users request. So the template can be totally broken Jinja, and we won't know it until the user is making requests against it.

Do you want to check the template at startup validity using the jinja library directly?

With the changes to transformers to add a default template to the tokenization_utils_base, there really isn't a case I could get it to crash due to a missing tokenizer. I am just going to remove the "else" entirely at this point.

I suppose i'll leave the code in place for:
```
parser.add_argument("--disable-endpoints",
                        type=json.loads,
                        default=[],
                        help="List of disabled endpoints")
```